TrafficCapture/captureKafkaOffloader/src/test/java/org/opensearch/migrations/trafficcapture/kafkaoffloader/KafkaCaptureFactoryTest.java

```diff
@@ -13,23 +21,76 @@
 import java.util.List;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
 
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.when;
+import static org.opensearch.migrations.trafficcapture.kafkaoffloader.KafkaCaptureFactory.KAFKA_MESSAGE_OVERHEAD_BYTES;
 
 @ExtendWith(MockitoExtension.class)
 public class KafkaCaptureFactoryTest {
 
     public static final String TEST_NODE_ID_STRING = "test_node_id";
     @Mock
     private Producer<String, byte[]> mockProducer;
+    private String connectionId = "0242c0fffea82008-0000000a-00000003-62993a3207f92af6-9093ce33";
+    private String topic = "test_topic";
 
-    private String connectionId = "test1234";
 
+    @Test
+    public void testLargeRequestIsWithinKafkaMessageSizeLimit() throws IOException, ExecutionException, InterruptedException {
+        final var referenceTimestamp = Instant.now(Clock.systemUTC());
+
+        int messageSize = 1024*1024;
+        MockProducer<String, byte[]> producer = new MockProducer<>(true, new StringSerializer(), new ByteArraySerializer());
+        KafkaCaptureFactory kafkaCaptureFactory =
+            new KafkaCaptureFactory(TEST_NODE_ID_STRING, producer, messageSize);
+        IChannelConnectionCaptureSerializer serializer = kafkaCaptureFactory.createOffloader(connectionId);
+
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < 15000; i++) {
+            sb.append("{ \"create\": { \"_index\": \"office-index\" } }\n{ \"title\": \"Malone's Cones\", \"year\": 2013 }\n");
+        }
+        Assertions.assertTrue(sb.toString().getBytes().length > 1024*1024);
+        byte[] fakeDataBytes = sb.toString().getBytes(StandardCharsets.UTF_8);
+        var bb = Unpooled.wrappedBuffer(fakeDataBytes);
+        serializer.addReadEvent(referenceTimestamp, bb);
+        CompletableFuture future = serializer.flushCommitAndResetStream(true);
+        future.get();
+        for (ProducerRecord<String, byte[]> record : producer.history()) {
+            int recordSize = calculateRecordSize(record, null);
+            Assertions.assertTrue(recordSize <= messageSize);
+            int largeIdRecordSize = calculateRecordSize(record, connectionId + ".9999999999");
+            Assertions.assertTrue(largeIdRecordSize <= messageSize);
+        }
+        bb.release();
+        producer.close();
+    }
+
+    /**
+     * This size calculation is based off the KafkaProducer client request size validation check done when Producer
+     * records are sent. This validation appears to be consistent for several versions now, here is a reference to
+     * version 3.5 at the time of writing this: https://github.com/apache/kafka/blob/3.5/clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java#L1030-L1032.
+     * It is, however, subject to change which may make this test scenario more suited for an integration test where
+     * a KafkaProducer does not need to be mocked.
```

Comment:
Thanks for the callout.  Do you have a follow up task for it?

Reply:
A follow up task for an integration test to cover this right?